From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mayagr <mayagr@qti.qualcomm.com>
Date: Tue, 28 Jul 2026 16:22:34 +0530
Subject: [PATCH 101/109] Cygwin: uname: report "aarch64" for
 PROCESSOR_ARCHITECTURE_ARM64

uname() and uname_x() returned "unknown" as the machine name on ARM64
Windows.  Map PROCESSOR_ARCHITECTURE_ARM64 to "aarch64".

(cherry picked from commit 5bc1d338f9febccc14ad8733b24b8842b5be8474)
(cherry picked from commit 51ecaca2f05279e10bdc7d53a7331d0a535c77f8)
(cherry picked from commit 4908d6ae72c159379feab0f4cb03a0f23aa0ca57)
---
 winsup/cygwin/uname.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/winsup/cygwin/uname.cc b/winsup/cygwin/uname.cc
index 8f984fac9..619ee800a 100644
--- a/winsup/cygwin/uname.cc
+++ b/winsup/cygwin/uname.cc
@@ -83,6 +83,9 @@ uname_x (struct utsname *name)
 	  case PROCESSOR_ARCHITECTURE_AMD64:
 	    strcpy (name->machine, "x86_64");
 	    break;
+	  case PROCESSOR_ARCHITECTURE_ARM64:
+	    strcpy (name->machine, "aarch64");
+	    break;
 	  default:
 	    strcpy (name->machine, "unknown");
 	    break;
@@ -165,6 +168,9 @@ uname (struct utsname *in_name)
 	  case PROCESSOR_ARCHITECTURE_AMD64:
 	    strcpy (name->machine, "x86_64");
 	    break;
+	  case PROCESSOR_ARCHITECTURE_ARM64:
+	    strcpy (name->machine, "aarch64");
+	    break;
 	  default:
 	    strcpy (name->machine, "unknown");
 	    break;
